File: 백준/Bronze/2798. 블랙잭/블랙잭.py
n, m = map(int, input().split())
arr = list(map(int, input().split()))

max_sum = 0
for i in range(n):
    for j in range(i + 1, n):
        for k in range(j + 1, n):
            sum = arr[i] + arr[j] + arr[k]
            if max_sum < sum <= m:
                max_sum = sum
print(max_sum)

File: 백준/Bronze/2798. 블랙잭/블랙잭.c
#include <stdio.h>

int main() {
	int n, m, sum = 0, max = 0;
	int arr[100] = { 0 };

	scanf("%d %d", &n, &m);
	for (int a = 0; a < n; a++) {
		scanf("%d", &arr[a]);
	}
	for (int i = 0; i < n; i++) {
		for (int j = i + 1; j < n; j++) {
			for (int k = j + 1; k < n; k++) {
				sum = arr[i] + arr[j] + arr[k];
				if ((max < sum) && (sum <= m))
					max = sum;
			}
		}
	}
	printf("%d", max);

	return 0;
}